### PR TITLE
KAFKA-5979: Use single AtomicCounter to generate internal names

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @dguy @bbejeck @norwood

**Comment by mjsax** [comment]
This fixes the regression. We should have caught this on the original PR already. The change to the test in this PR, reverts the change to the test of the original one. This should have been an indicator that something is wrong. Thus, no need to add a new test IMHO.

**Comment by norwood** [comment]
😍

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java [hunk 1]
```diff
@@ -159,7 +159,7 @@ public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuil
         return new GlobalKTableImpl<>(new KTableSourceValueGetterSupplier<K, V>(storeBuilder.name()));
     }
 
-    String newName(final String prefix) {
+    public String newName(final String prefix) {
```
Comment (bbejeck) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java:162]: Put a TODO to remove `public` once we remove `KStreamBuilder`?

**Comment by bbejeck** [comment]
Thanks, @mjsax. I agree with you on the test.  
Just one comment, otherwise LGTM.

**Review by dguy — APPROVED** [review]
Thanks @mjsax, LGTM

**Comment by dguy** [comment]
merged to trunk

**Pull request closed without merging**